### PR TITLE
Detected language in search dates return

### LINKED ISSUE #396: Add detected language to return value of search_dates()
E.g.:
```
In:  search_dates('You can reach me on the phone')
Out: [('me', datetime.datetime(2018, 3, 14, 0, 0), 'fr')]
```

Referred in: https://github.com/scrapinghub/dateparser/issues/394

### PR DESCRIPTION
**Pull request by rennerocha**
This should fix #396 .

### CODE REVIEW

File: dateparser/search/__init__.py [hunk 1]
```diff
@@ -1,35 +1,55 @@
...
-    result = _search_with_detection.search_dates(text=text, languages=languages, settings=settings)
-    if result['Dates']:
-        return result['Dates']
+    result = _search_with_detection.search_dates(
+        text=text, languages=languages, settings=settings
+    )
+    language, dates = result.get('Language'), result.get('Dates')
+    if dates:
+        if add_detected_language:
+            dates = [date + (normalize_unicode(language), ) for date in dates]
```
Comment (asadurski) [inline dateparser/search/__init__.py]: Can you give a hint why `normalize_unicode()` is necessary here? Do we have any irregular language shortnames?

File: dateparser/search/__init__.py [hunk 2]
```diff
@@ -1,35 +1,55 @@
...
 
+        >>> from dateparse.search import search_dates
         >>> search_dates('The first artificial Earth satellite was launched on 4 October 1957.')
         [('on 4 October 1957', datetime.datetime(1957, 10, 4, 0, 0))]
 
+        >>> search_dates("The client arrived to the office for the first time in March 3rd, 2004 and got serviced, after a couple of months, on May 6th 2004, the customer returned indicating a defect on the part")
+        [(u'in March 3rd, 2004 and', datetime.datetime(2004, 3, 3, 0, 0)),
+        (u'on May 6th 2004', datetime.datetime(2004, 5, 6, 0, 0))]
+
+        >>> search_dates('The first artificial Earth satellite was launched on 4 October 1957.')
```
Comment (asadurski) [inline dateparser/search/__init__.py]: This should include `add_detected_language=True`, right?

File: dateparser/search/__init__.py (hunk 1 above)
Comment (asadurski) [inline dateparser/search/__init__.py]: I'm not sure what would be the best output. Currently Dateparser detects only one language for entire input text and that would call for `([list of dates], detected_language)` output—but if we go for more detailed segmentation or language selection, the way you did it really makes sense.
Yeah, I think we should keep it this way.

File: dateparser/search/__init__.py [hunk 3]
```diff
@@ -1,35 +1,55 @@
...
-        :return: Returns list of tuples containing pairs:
-                 substrings representing date and/or time and corresponding :mod:`datetime.datetime` object.
+        :return: Returns list of tuples containing:
+                 substrings representing date and/or time, corresponding :mod:`datetime.datetime`
+                 object and detected language if *add_detected_language* is True.
                  Returns None if no dates that can be parsed are found.
         :rtype: list
         :raises: ValueError - Unknown Language
 
+        >>> from dateparse.search import search_dates
```
[the remark this answers is not in the document]
Comment (rennerocha) [reply dateparser/search/__init__.py]: Fixed!

File: dateparser/search/__init__.py (hunk 1 above)
Comment (rennerocha) [reply dateparser/search/__init__.py]: It is not necessary! I removed it.

File: dateparser/search/__init__.py (hunk 2 above)
Comment (rennerocha) [reply dateparser/search/__init__.py]: Fixed!

**Comment by asadurski** [comment]
Nice, thank you!

**Pull request merged**